cg/apps/cgstats/crud/find.py

```diff
@@ -23,52 +23,83 @@
 class FindHandler:
     def get_support_parameters_id(self, demux_results: DemuxResults) -> Optional[int]:
         """Fetch the id of the support parameters if post exists"""
-        LOG.debug("Search for support parameters with file %s", demux_results.results_dir)
-        support_parameters_id: Optional[int] = Supportparams.exists(str(demux_results.results_dir))
-        if support_parameters_id:
-            LOG.debug("Found support parameters with id %s", support_parameters_id)
+        LOG.debug(f"Search for support parameters with file {demux_results.results_dir}")
+
+        support_parameters: Supportparams = self.get_support_parameters_by_document_path(
+            document_path=str(demux_results.results_dir)
+        )
+
+        if support_parameters:
+            support_parameters_id = support_parameters.supportparams_id
+            LOG.debug(f"Found support parameters with id {support_parameters_id}")
             return support_parameters_id
+
         LOG.debug("Could not find support parameters")
         return None
 
+    def get_support_parameters_by_document_path(
+        self, document_path: str
+    ) -> Optional[Supportparams]:
+        return Supportparams.query.filter_by(document_path=document_path).first()
+
     def get_datasource_id(self, demux_results: DemuxResults) -> Optional[int]:
         """Fetch the datasource id for a certain run"""
         stats_path = {
             "bcl2fastq": demux_results.conversion_stats_path,
             "dragen": demux_results.demux_stats_path,
         }
-        LOG.debug("Search for datasource with file %s", stats_path[demux_results.bcl_converter])
-        datasource_id: Optional[int] = Datasource.exists(
-            str(stats_path[demux_results.bcl_converter])
-        )
-        if datasource_id:
-            LOG.debug("Found datasource with id %s", datasource_id)
-            return datasource_id
+        document_path: str = str(stats_path[demux_results.bcl_converter])
```

Comment:
No need to cast to string

Reply:
`stats_path[demux_results.bcl_converter]` returns a Path, I think I prefer converting to a string instead of passing a Path object